store-implementation/parquet-store/src/main/java/uk/gov/gchq/gaffer/parquetstore/operation/handler/spark/utilities/WriteData.java

```diff
@@ -56,28 +59,47 @@ public WriteData(final Function<String, String> groupToDirectory, final Schema s
     public void call(final Iterator<Element> elements) throws Exception {
         final SchemaUtils schemaUtils = new SchemaUtils(Schema.fromJson(schemaAsJson));
         final int partitionId = TaskContext.getPartitionId();
+        final long taskAttemptId = TaskContext.get().taskAttemptId();
         final Map<String, ParquetWriter<Element>> groupToWriter = new HashMap<>();
+        final Map<String, Path> groupToWriterPath = new HashMap<>();
+
         for (final String group : schemaUtils.getGroups()) {
-            groupToWriter.put(group,
-                    buildWriter(group, groupToDirectory.get(group) + "/input-" + partitionId + ".parquet", schemaUtils));
+            groupToWriterPath.put(group, new Path(groupToDirectory.get(group) + "/input-" + partitionId + "-" + taskAttemptId + ".parquet"));
```

Comment:
Can you add a unit test for this please?

Reply:
There already are unit tests that the process works, and those still pass. The issue only happens when speculative execution causes multiple attempts of the same task on the same partition. I don't see how it's possible to unit test that scenario. (The title of this PR is misleading - changing the name of the file to include the attempt id is just part of the solution, not the point of the PR). But it is probably possible to test two manual attempts to process the same partition and check that only one works - I'll add that.